### PR TITLE
BGP neighbors getters: hail for vendor consistency

### PR DESCRIPTION
**Pull request by mirceaulinic**
The peer_fwd_rti field we rely on to identify the RTI
is not widely adopted on all Junos platforms, although it has been
introduced theoretically beginning with Junos 15. Yay consistency!
That means:
- we solve #191
- the execution will be slower, as we firstly fetch the list of
routing instances, then extract the neighbors under each and
every instance.

### CODE REVIEW

**Comment by dbarrosop** [comment]
Why are you removing a test case?

File: napalm_junos/junos.py [hunk 1]
```diff
@@ -999,23 +999,23 @@ def _bgp_iter_core(neighbor_data, instance=None):
...
+        instances = junos_views.junos_route_instance_table(self.device)
+        for instance, instance_data in instances.get().items():
+            if instance.startswith('__'):
+                # junos internal instances
+                continue
+            neighbor_data = bgp_neighbors_table.get(instance=instance,
+                                                    neighbor_address=neighbor_address).items()
+            _bgp_iter_core(neighbor_data, instance=instance)
+        # else:
+        #     bgp_neighbors_table = junos_views.junos_bgp_neighbors_table(self.device)
```
Comment (dbarrosop) [inline napalm_junos/junos.py:1131]: Why all this commented code? Shouldn't we just remove it?

Comment (mirceaulinic) [reply napalm_junos/junos.py:1131]: I thought about leaving them for later versions when hopefully the `peer_fwd_rti` key will be available on all Junos distributions.
But we can remove, no problem.

Comment (dbarrosop) [reply napalm_junos/junos.py:1131]: As discussed on slack. I am fine with the comments. I'd suggest though to either add an explantion to the comments (a comment for the comments :P) to give context to other people or remove them, whatever you prefer.

Comment (mirceaulinic) [reply napalm_junos/junos.py:1131]: Added comments: https://github.com/napalm-automation/napalm-junos/pull/193/commits/0d02e074915e6e94051381c95b6a133bdf85857b

**Review by dbarrosop — APPROVED** [review]

**Pull request merged**